airflow/providers/amazon/aws/triggers/glue_session.py

```diff
@@ -0,0 +1,59 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from typing import Any, AsyncIterator
+
+from airflow.providers.amazon.aws.hooks.glue_session import GlueSessionHook
+from airflow.triggers.base import BaseTrigger, TriggerEvent
+
+
+class GlueSessionReadyTrigger(BaseTrigger):
+    """
+    Watches for a glue session, triggers when it is ready.
+
+    :param session_id: glue session id
+    :param aws_conn_id: The Airflow connection used for AWS credentials.
+    """
+
+    def __init__(
+        self,
+        session_id: str,
+        aws_conn_id: str | None,
```

Comment:
There is also common hook settings exists: `region_name`, `verify` and `botocore_config` ?